### DISCUSSION TITLE
Weird Plots

### DISCUSSION THREAD

**Discussion by danilomoggia** [cdriveraus/ctsem#24] [Q&A]
Hi Charly,
I computed the cross-lagged effect for a specific group of subjects and I obtained a plot with a very solid painting of credible intervals (instead of the traditional one more transparent). Here I'm pasting the code and the plot. Is there any reason for this?

ctStanDiscretePars(SRP_ModelFit, plot=T, subjects= c(9, 13, 18, 21, 27, 35, 43, 44, 56),
                   facets='Effect', indices = 'CR',
                   cores = 5)

![image](https://user-images.githubusercontent.com/85121548/144409286-f5d76192-865c-41d3-a9e7-98b1a938edfc.png)

Many thanks!

Danilo.

**Comment by cdriveraus** [comment]
It kind of looks like you don't have individual differences in the coefficients, and so there are multiple plots overlapping each other?

**Reply by danilomoggia** [reply]
Interesting! It could be, but I'm not sure because these subjects belong to the same subgroup within the sample. How can I check that?
Within the sample, there are 3 subgroups. I dummy coded the grouping variable, and I introduced it as a time-independent predictor. I haven't put any special configuration in any of the matrices (I'm using default parameters). Could it be that? I recognise that I get a bit lost with the matrix configuration (except for the lambda matrix) because I'm not sure which parameters fix or freely estimate on which purpose.

**Reply by cdriveraus** [reply]
yes, so you have 'deterministic' variation, defined by the effect of the group, but by default ctsem only has random variation on intercept terms. you can see this in the indvarying column of mymodel$pars and also in the subject distribution shown when using ctModelLatex(mymodel)

**Reply by danilomoggia** [reply]
It makes sense! Thanks Charly! Is it possible to configure variation within groups? My model is this one:

SRP_Model <- ctModel(type = "stanct", manifestNames = c("S", "R", "P"),latentNames = c("S", "R", "P"),
                      n.TIpred = 2, TIpredNames = c("TI1", "TI2"), LAMBDA = diag(3),
                     tipredDefault = FALSE)

Many thanks again!

**Reply by cdriveraus** [reply]
yes, you can modify the mymodel$pars$indvarying column as described, or specify the parameter matrices explicity and use the | separator notation to denote which parameters have random effects. Section 2 of the manual describes this kind of thing -- https://cran.r-project.org/web/packages/ctsem/vignettes/hierarchicalmanual.pdf
Lots of random effects will make estimation and interpretation a bit trickier.

**Reply by danilomoggia** [reply]
Great! Thank you very much!
